### PR TITLE
Add a superclass reference to class descriptors

### PR DESCRIPTION
**Pull request by rjmccall**
And some miscellaneous related improvements.

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

File: include/swift/Runtime/Metadata.h [hunk 1]
```diff
@@ -3061,16 +3065,9 @@ class TargetClassDescriptor final
...
-  /// if this is a class descriptor with a vtable descriptor for runtime
-  /// vtable instantiation.
-  static constexpr const uint16_t HasVTableFlag =
-    uint16_t(TypeContextDescriptorFlags::HasVTable);
-  /// This bit is set in the context descriptor header's kind-specific flags
-  /// if this is a class descriptor with a resilient superclass.
-  static constexpr const uint16_t HasResilientSuperclassFlag =
-    uint16_t(TypeContextDescriptorFlags::HasResilientSuperclass);
-  
+  /// The superclass of this class.  To interpret this pointer,
```
Comment (jckarter) [inline include/swift/Runtime/Metadata.h]: Comment's cut off here.

Comment (rjmccall) [reply include/swift/Runtime/Metadata.h]: Oops.  I'll fix in this PR if I have to submit again, otherwise it'll go in the next one.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test OS X.

**Comment by rjmccall** [comment]
@swift-ci Please test OS X.

**Comment by rjmccall** [comment]
AFAICT, the last test run succeeded, but there was a connection failure (that only shows up in the smoke-test log?), ergo kaboom.

**Pull request merged**